rocksdb_replicator/replicated_db.cpp

```diff
@@ -141,6 +148,42 @@ RocksDBReplicator::ReplicatedDB::ReplicatedDB(
           + FLAGS_replicator_client_server_timeout_difference_ms));
 }
 
+/*
+ * Helper function to reset the upstream IP after querying for latest leader from helix.
+ */
+void RocksDBReplicator::ReplicatedDB::resetUpstream(std::shared_ptr<ReplicatedDB> db) {
+  if (FLAGS_replicator_zk_cluster.empty() || FLAGS_replicator_helix_cluster.empty()) {
+    LOG(ERROR) << "[resetUpstream] ZK cluster or helix cluster name not provided.";
+  }
+
+  // Query helix only FLAGS_replication_error_reset_upstream_percentage %ge of the requests.
+  // This is to avoid excessive load to Helix (zk)
+  if (common::GetRandInt(0, 99) >= FLAGS_replication_error_reset_upstream_percentage) {
+    LOG(ERROR) << "[resetUpstream] Skip checking latest upstream for " << db->db_name_;
```

Comment:
when replication_error happen, what kind of rate of error is? 
Will this write too many error logs into logfile?

Reply:
Error rate is quite high, but once an error is seen it is not retried for 5 seconds, so I think this should be okay.